src/modules/epsr/process.cpp

```diff
@@ -293,10 +298,22 @@ Module::ExecutionResult EPSRModule::process(ModuleContext &moduleContext)
         // Calculate r-factor over fit range and store
         auto tempRefData = originalReferenceData;
         Filters::trim(tempRefData, qMin_, qMax_);
-        auto rFactor = Error::rFactor(tempRefData, weightedSQ.total()).error;
+        auto errorReport = Error::rFactor(tempRefData, weightedSQ.total());
+        lowerBound = errorReport.firstX;
+        upperBound = errorReport.lastX;
+        auto rFactor = errorReport.error;
         rFacTot += rFactor;
         errors.addPoint(moduleContext.dissolve().iteration(), rFactor);
-        Messenger::print("Current R-Factor for reference data '{}' is {:.5f}.\n", module->name(), rFactor);
+        Messenger::print("Current R-Factor for reference data '{}' over complete range ({:.5f} to {:.5f}) is {:.5f}.\n",
+                         module->name(), lowerBound, upperBound, rFactor);
+        // Calculate r-factor over specified ranges_
+        for (auto &&[range, rangeTot] : zip(ranges_, rangedRFacTots))
+        {
+            rFactor = Error::rFactor(tempRefData, weightedSQ.total(), range).error;
+            rangeTot += rFactor;
+            Messenger::print("Current R-Factor for reference data '{}' over range {:.5f} to {:.5f} is {:.5f}.\n",
+                             module->name(), range.minimum(), range.maximum(), rFactor);
```

Comment:
Regarding this piece of code, here we report the range bounds as defined, but above we reported the first and last points considered. Should we change this to report the actual data range over which the error was calculated to ensure there is no mistaking what the value represents?
```suggestion
            const auto rFactorRangeReport = Error::rFactor(tempRefData, weightedSQ.total(), range).error;
            rangeTot += rFactorRangeReport.error;
            Messenger::print("Current R-Factor for reference data '{}' over range {:.5f} to {:.5f} is {:.5f}.\n",
                             module->name(), range.minimum(), range.maximum(), rFactorRangeReport.error);
```

Reply:
So do you mean to report it as:
```
Messenger::print("Current R-Factor for reference data '{}' over range {:.5f} to {:.5f} is {:.5f}.\n",
                             module->name(), rFactorRangeReport.firstX, rFactorRangeReport.lastX, rFactorRangeReport.error);
```

Just for further clarification, the reason I decided to include the data's total range in the above report was that I thought it would be helpful for people to know the range of the whole data, as it might help with selecting ranges.